src/Features/LanguageServer/ProtocolUnitTests/CodeActions/CodeActionsTests.cs

```diff
@@ -201,11 +201,8 @@ private void XYZ()
         };
 
         var results = await RunGetCodeActionsAsync(testLspServer, codeActionParams);
-        var resultsTitles = results.Select(r => r.Title).ToArray();
-        // Inline method refactoring provide nested code actions.
-        // Make sure it is correctly displayed.
-        Assert.True(resultsTitles.Contains("Inline 'A()' -> Inline 'A()'"));
-        Assert.True(resultsTitles.Contains("Inline 'A()' -> Inline and keep 'A()'"));
+        // Assert that nested code actions aren't enumerated.
+        Assert.Equal(6, results.Length);
```

Comment:
Can you add a few more tests for these scenarios
1.  One that verifies the structure of the nested code actions (and that it returns the correct structure for multiple levels of nesting, has all the right command names, etc)
2.  One that verifies the structure in the presence of a fix all + nested action (if that exists)
3.  One that verifies resolve on the nested code action does nothing.
4.  One that verifies we don't get these in VS (and instead get the VS code action types)